pyomo/contrib/doe/doe.py

```diff
@@ -25,1023 +25,749 @@
-            A 2D numpy array containing Fisher information matrix (FIM) for prior experiments.
-            The default None means there is no prior information.
-        discretize_model:
-            A user-specified ``function`` that discretizes the model. Only use with Pyomo.DAE, default=None
-        args:
-            Additional arguments for the create_model function.
+            2D numpy array representing information from prior experiments. If no value is given,
+            the assumed prior will be a matrix of zeros. This matrix will be assumed to be scaled
+            as the user has specified (i.e., if scale_nominal_param_value is true, we will assume
+            the FIM provided here has been scaled by the parameter values)
+        jac_initial:
+            2D numpy array as the initial values for the sensitivity matrix.
+        fim_initial:
+            2D numpy array as the initial values for the FIM.
+        L_diagonal_lower_bound:
+            Lower bound for the values of the lower triangular Cholesky factorization matrix.
+            default: 1e-7
+        solver:
+            A ``solver`` object specified by the user, default=None.
+            If not specified, default solver is set to IPOPT with MA57.
+        tee:
+            Solver option to be passed for verbose output.
+        get_labeled_model_args:
+            Additional arguments for the ``get_labeled_model`` function on the Experiment object.
+        _Cholesky_option:
+            Boolean value of whether or not to use the cholesky factorization to compute the
+            determinant for the D-optimality criteria. This parameter should not be changed
+            unless the user intends to make performance worse (i.e., compare an existing tool
+            that uses the full FIM to this algorithm)
+        _only_compute_fim_lower:
+            If True, only the lower triangle of the FIM is computed. This parameter should not
+            be changed unless the user intends to make performance worse (i.e., compare an
+            existing tool that uses the full FIM to this algorithm)
         logger_level:
             Specify the level of the logger. Change to logging.DEBUG for all messages.
-        only_compute_fim_lower:
-            If True, only the lower triangle of the FIM is computed. Default is True.
         """
+        # Deprecation error
+        if 'create_model' in kwargs:
+            raise ValueError(
+                "DEPRECATION ERROR: Pyomo.DoE has been refactored. The current interface utilizes Experiment "
+                "objects that label unknown parameters, experiment inputs, experiment outputs and measurement "
+                "error. This avoids string-based naming which is fragile. For instruction to use the new "
+                "interface, please see the Pyomo.DoE under the contributed packages documentation at "
+                "`https://pyomo.readthedocs.io/en/latest/contributed_packages/doe/doe.html`"
+            )
```

Comment:
There is a utility for this:
```
from pyomo.common.deprecation import deprecation_warning
deprecation_warning(message, version='6.7.4.dev0')
```

Reply:
Does the warning cause the code to stop? Or do I need to include an additional error?

Reply:
You would need to include a `raise` statement if the idea is to get the code to stop at this point, yes.